terraform/variables/README.md

```diff
@@ -0,0 +1,205 @@
+real-world use-case. It creates static Terraform output variables of various types (string, list, and map), and then references
+them in local docker deployment config.
+
+## Terraform Setup
+
+### Org and Workspace Creation
+
+First, we'll need a Terraform Cloud workspace. If you don't have one already, you can create one
+using the Terraform found in `terraform/tfc`. If you already have a Terraform org and workspace,
+skip ahead to the next section.
+
+This module requires three inputs:
+
+- A TFC API token. Follow [this guide](https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/api-tokens)
+to obtain a TFC api token.
+
+- A TFC organization name. This is arbitrary, but needs to be globally unique.
+
+- An email address to use as your TFC org admin contact address.
+
+Run Terraform to create the TFC org and workspace. You'll be prompted for the above inputs
+
+```shell
+$ cd ./terraform/tfc
+$ terraform init
+...
+$ terraform apply
+...
+Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
+```
+
+### Generating Terraform Outputs
+
+In `terraform/main`, we have the minimal possible Terraform to generate ouputs. This is meant to simulate
+real Terraform that might be creating and AWS VPC, and outputting things like VPC IDs and subnet arns.
+
+You'll need the following inputs:
+
+- An org ID and workspace ID. These cannot be referenced as variables, so you'll need to modify 
+`./terraform/main/main.tf`. Use the organization name chosen in the previous step, or set this
+to your own organization ID if desired.
+
+```hcl
+terraform {
+  cloud {
+    organization = "waypoint_tfc_vars_example" // <- you must change this
+
+    workspaces {
+      name = "waypoint_tfc_vars_example"
+    }
+  }
+}
+```
+
+Run the Terraform to produce the outputs:
+```shell
+$ cd ./terraform/main
+$ terraform init
+...
+$ terraform apply
+...
+Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+list_of_strings = [
+  "item_1",
+  "item_2",
+  "item_3",
+]
+map_of_strings = {
+  "key1" = "value1"
+  "key2" = "value2"
+}
+simple_string = "justastring"
+```
+
+## Waypoint setup
+
+#### Waypoint CLI
+
+Install the Waypoint CLI. See [installation instructions](https://developer.hashicorp.com/waypoint/tutorials/get-started-docker/get-started-install).
+
+#### Waypoint Server
+
+You'll need a Waypoint server. We recommend using [HCP Waypoint](https://cloud.hashicorp.com/products/waypoint)
+You'll also need your CLI configured with a [context](https://developer.hashicorp.com/waypoint/commands/context-create)
+pointing to this server.
+
+#### Waypoint remote runner
+
+Waypoint remote runners are required to use variables with dynamic defaults. HCP Waypoint will 
+walk you through installing a remote runner into your infrastructure. If you don't have
+a runner already, you can run `waypoint runner agent` in a separate terminal window. This
```

Comment:
Maybe we should also suggest running the `waypoint runner install` command too?

Reply:
This example uses the docker deploy plugin, and I don't think of the platforms that `waypoint runner install` supports gives the runner direct access to a docker daemon. The rest of the example also assumes that the deployed docker container will run on the dev machine, and only a local runner can pull that off.